pkg/resources/apiserver/deployment.go

```diff
@@ -90,11 +90,17 @@ func DeploymentCreator(data *resources.TemplateData, enableOIDCAuthentication bo
 
 			etcdEndpoints := etcd.GetClientEndpoints(data.Cluster().Status.NamespaceName)
 
-			// Configure user cluster DNS resolver for this pod.
-			dep.Spec.Template.Spec.DNSPolicy, dep.Spec.Template.Spec.DNSConfig, err = resources.UserClusterDNSPolicyAndConfig(data)
-			if err != nil {
-				return nil, err
+			if !data.IsKonnectivityEnabled() {
+				dep.Spec.Template.Spec.DNSPolicy, dep.Spec.Template.Spec.DNSConfig, err = resources.UserClusterDNSPolicyAndConfig(data)
+				if err != nil {
+					return nil, err
+				}
+			} else {
+				// custom DNS resolver in not needed in Konnectivity setup
+				dep.Spec.Template.Spec.DNSPolicy = corev1.DNSClusterFirst
+				dep.Spec.Template.Spec.DNSConfig = nil
```

Comment:
nit: This `if...else...` would easier to read if it was the other way round.

`   `if data.IsKonnectivityEnabled() { ...} else {...}`